src/main/java/jp/co/dwango/logback/SlackWebhookAppender.java

```diff
@@ -1,167 +1,434 @@
-        this.username = username;
+    
+    /**
+     * Sets timeout
+     * 
+     * @param timeout timeout
+     */
+    public void setTimeout(String timeout) {
+        this.timeout = Integer.valueOf(timeout);
     }
 
-    public String getIconEmoji() {
-        return this.iconEmoji;
+    /**
+     * Sets timeout
+     * 
+     * @param timeout timeout
+     */
+    public void setTimeout(int timeout) {
+        this.timeout = timeout;
     }
 
-    public void setIconEmoji(String iconEmoji) {
-        this.iconEmoji = iconEmoji;
-    }
-
-    public String getIconUrl() {
-        return this.iconUrl;
-    }
-
-    public void setIconUrl(String iconUrl) {
-        this.iconUrl = iconUrl;
-    }
-
-    public boolean getLinkNames() {
-        return this.linkNames;
-    }
+    /**
+     * @see ch.qos.logback.core.UnsynchronizedAppenderBase#start()
+     */
+    @Override
+    public void start() {
+        // backward compatibility mode
+        if(this.payload == null) {
+            bc.start();
+            return;
+        }
+        
+        int errors = 0;
+        try {
+            initScriptEngine();
+        } catch (Exception e) {
+            errors++;
+            addError(e.getMessage(), e);
+        }
 
-    public void setLinkNames(boolean linkNames) {
-        this.linkNames = linkNames;
-    }
+        if (webhookUrlStr == null) {
+            errors++;
+            addError("Webhook URL is not specified.");
+        } else {
+            try {
+                this.webhookUrl = new URL(webhookUrlStr);
+            } catch (MalformedURLException e) {
+                errors++;
+                addError("Webhook URL is not malformed.", e);
+            }
+        }
 
-    public Layout<ILoggingEvent> getLayout() {
-        return this.layout;
+        if (errors == 0) {
+            super.start();
+        }
     }
-
-    public void setLayout(Layout<ILoggingEvent> layout) {
-        this.layout = layout;
+    
+    /**
+     * Initializes a {@link ScriptEngine}
+     * 
+     * @throws ScriptException if an error occurrs in payload script.
+     * @throws IllegalStateException if {@link #payload} field is null or empty
+     */
+    private void initScriptEngine() throws ScriptException {
+        String payload = this.payload;
+        if(payload == null) {
+            throw new IllegalStateException("Pyaload is null");
+        }
+        payload = payload.trim();
+        if(payload.isEmpty()) {
+            throw new IllegalStateException("Pyaload is empty");
+        }
+        
+        // looks up and creates JavaScript engine
+        this.engine = SCRIPT_MANAGER.getEngineByName("js");
+        
+        // bind hostname
+        Bindings bindings = engine.getBindings(ScriptContext.ENGINE_SCOPE);
+        bindings.put("hostname", resolveHostname());
+        
+        // define formatTimestamp function
+        engine.eval(
+            "function formatTimestamp(timestamp) { " +
```

Comment:
You can format timestamp by specifying `<pattern>%date{ISO8601}</pattern>` to your logback.xml.